api/v1beta2/cryostat_types.go

```diff
@@ -487,12 +488,38 @@ type TemplateConfigMap struct {
 
 // Authorization options provide additional configurations for the auth proxy.
 type AuthorizationOptions struct {
-	// Reference to a secret and file name containing the Basic authentication htpasswd file
+	// Configuration for OpenShift RBAC to define which OpenShift user accounts may access the Cryostat application.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	OpenShiftSsoRoles *OpenShiftSsoRoles `json:"openShiftSsoRoles,omitempty"`
+	// Reference to a secret and file name containing the Basic authentication htpasswd file. If deploying on OpenShift this
+	// defines additional user accounts that can access the Cryostat application, on top of the OpenShift user accounts which
+	// pass the OpenShiftSsoRoles checks. If not on OpenShift then this defines the only user accounts that have access.
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
 	BasicAuth *SecretFile `json:"basicAuth,omitempty"`
 }
 
+type OpenShiftSsoRoles struct {
+	// Disable OpenShift SSO integration and allow all users to access the application without authentication. This
+	// will also bypass the BasicAuth, if specified.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Disable OpenShift SSO",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	Disabled *bool `json:"disabled,omitempty"`
+	// The SubjectAccessReview that interactive clients (users visiting the application via web browser) must pass in
+	// order to access the application. If not specified, the default role required is "create pods/exec" in the Cryostat
+	// application's installation namespace.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	SubjectAccessReview *authzv1.ResourceAttributes `json:"subjectAccessReview,omitempty"`
```

Comment:
Should this accept multiple ResourceAttributes? Then the user would need to have all permissions listed. I think this would be equivalent to the current role-based system in 2.4.